docs/content/src/develop/running-as-a-java-program.md

```diff
@@ -1,105 +1,119 @@
+Add the following to the runtime config:
+@encrypted:{pIQrB9YfCQK1eIWH5d6UaZXA3zr+60JxSBcpa2PY7a8=}
 
-To add Docker support, add the following code to the service you created above.
+Or add to the runtime command line:
+-e<param>=@encrypted:{pIQrB9YfCQK1eIWH5d6UaZXA3zr+60JxSBcpa2PY7a8=}
+```
+
+Ballerina uses AES, CBC mode with PKCS#5 padding for encryption. The generated encrypted value should be used in place of the plain-text configuration value.
+
+For example, contents of a configuration file that includes a secret value should look as follows:
+
+```
+api.secret="@encrypted:{pIQrB9YfCQK1eIWH5d6UaZXA3zr+60JxSBcpa2PY7a8=}"
+api.provider="not-a-security-sensitive-value"
+```
+
+When running a Ballerina program that uses encrypted configuration values, it is required to provide the secret used during the encryption process to perform the decryption.
+
+A file named `secret.txt` is used for this purpose. If such file exists, the decryption secret is read from the file and immediately removed from the file to make sure secret cannot be accessed afterwards.
+
+The file based approach is useful in automated deployments. The file containing the decryption secret can be deployed along with the Ballerina program. The name and the path of the secret file can be configured using the `ballerina.config.secret` runtime parameter:
+
+```java
+
+$ java -jar program.jar —ballerina.config.secret=src/MyModule/resources/secret.txt
+
+```
+
+## Passing configurations as environment variables
+
+If the values are not set in the config file, it is read from the environment variables. For example, if you set a=10 in the command line as an environment variable and if your program's configuration is reading config key a, it will read it as 10 even if the a value is set differently in the config file. So priority is given to the environment variable you set.
+
+Consider the following example, which reads a Ballerina config value and prints it.
 
 ```ballerina
-import ballerina/http;  
-import ballerinax/docker;  
-  
-@http:ServiceConfig {  
-    basePath:"/helloWorld"  
-}  
-@docker:Config {
-    registry:"docker.abc.com",
-    name:"helloworld",
-    tag:"v1.0"
-}
-service helloWorld on new http:Listener(9090) {
-    resource function sayHello (http:Caller caller, http:Request request) {
-        http:Response response = new;
-        response.setTextPayload("Hello, World! \n");
-        _ = caller -> respond(response);
-    }
+import ballerina/io;
+import ballerina/config;
+
+public function main() {
+  string name = config:getAsString("hellouser");
+  io:println("Hello, " + name + " !");
 }
 ```
 
-Now your code is ready to generate deployment artifacts. In this case it is a Docker image.
-  
-```bash
-$ ballerina build hello_world_docker.bal  
-Compiling source
-    hello_world_docker.bal
+The config key is `hellouser`. To pass a value to this config from the CLI, we can run the following command. The `-e` argument passes the key and value to the program.
```

Comment:
where is the usage of `-e` tag in the below command?

Reply:
Removed this line. This is not required it seems.